test/tools/WebListener/Controllers/LinkController.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Primitives;
+using mvc.Models;
+
+namespace mvc.Controllers
+{
+    public class LinkController : Controller
+    {
+        public JsonResult Index()
+        {
+            StringValues maxLinksSV;
+            int maxLinks;
+            if(!Request.Query.TryGetValue("maxlinks", out maxLinksSV) || !Int32.TryParse(maxLinksSV.FirstOrDefault(), out maxLinks) || maxLinks < 1)
```

Comment:
As far as I know, one of the last .Net Core SDK updates included the latest version of Roslyn, therefore it should be possible to inline the out parameter declarations as e.g.  `out StringValues maxLinksSV`, etc.
Very minor but here and below is no space after the `if` statement (tip: just select everything and `Ctrl+K+F` auto-formats you all those things)

Reply:
I'm not really a C# developer... so that inline trick is new to me. Good to know! I've changed all the instances of it.

For some reason.. the keyboard shortcuts for formatting are not working for me *scratches head. I had to right-click and chose format document. weird. Any way, formatting is fixed now.

Reply:
I know, C# is moving quite fast now and it is hard to keep up, you can have a read [here](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-2) or watch one of the videos from Kasey Uhlenhuth if you are interested but I mainly learn those tricks also only from other people. Otherwise here are my personal favourites :
- [String interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-6#string-interpolation) 
- [Expression-bodied function members](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-6#expression-bodied-function-members)
- [Tuple naming instead of Item1, Item2](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#tuples)

Reply:
Thanks for pointing me in the right direction.  Discards will come in handy too.